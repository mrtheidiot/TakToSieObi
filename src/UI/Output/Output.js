import React from 'react'
import { Link } from 'react-router-dom'

import classes from './Output.module.css'

const Output = (props) => {

  const cos1 = props.text.split("/addresses/");
  const cos2 = cos1[1] ? cos1[1].split(",") : ""
  const coslink = cos1[1] ? cos1[1] : ""

  return (
    <> 
    <ul className={classes.text}>
      {cos1[0].split("/nl/").map((item, index) => (
        <div key={index}>
          {item.split("/b/").map((item2, index) => (
              <ul key={index}>{index & (2 !== 0) ? <b>{<ul>{item2.split("/l/").map((item3,index) => <li key={index}>
              {index % 2 !== 0 ? <Link to={coslink}>{item3}</Link> : item3}
            </li>)}</ul>}</b> : <ul>{item2.split("/l/").map((item3,index) => <li key={index}>
                          {index % 2 !== 0 ? <a href={coslink}>{item3}</a> : item3}
                        </li>)}</ul>}</ul>
          ))}
        </div>
      ))}
    </ul>
    </>
  );
}

export default Output