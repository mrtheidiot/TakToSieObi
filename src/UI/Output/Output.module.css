.text {
    display: inline;
}

.text ul {
    display: inline;
}

.text ul li {
    display: inline;
}