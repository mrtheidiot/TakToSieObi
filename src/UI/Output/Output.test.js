import { render } from "@testing-library/react";

import Output from "./Output";

it("matches the Output snapshot", () => {
  const text =
    "Is it /b/possible/b/ to be such a /l/beaurifull/l/ person?/addresses//treningi";

  const {
    container: { firstChild },
  } = render(<Output text={text} />);

  expect(firstChild).toMatchSnapshot();
});
