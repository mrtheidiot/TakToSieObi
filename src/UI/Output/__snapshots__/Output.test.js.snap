// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Output snapshot 1`] = `
<div
  class="text"
  data-testid="output"
>
  <div>
    <div>
      <div>
        <div>
          Is it 
        </div>
      </div>
    </div>
    <div>
      <b>
        <div>
          <div>
            possible
          </div>
        </div>
      </b>
    </div>
    <div>
      <div>
        <div>
           to be such a 
        </div>
        <div>
          <a
            href="/treningi"
          >
            beaurifull
          </a>
        </div>
        <div>
           person?
        </div>
      </div>
    </div>
  </div>
</div>
`;
